src/components/StructBuilder.js

```diff
@@ -185,26 +193,26 @@ export const StructBuilder = props => {
                         setSource: setInnerStructSource,
                         dataTableAttributes
                       }),
-                      warningMessage: missingExpectedAttributes.includes(structInputDefinition[rowIndex].field_name) ? 'This attribute doesn\'t exist in the data table' : ''
+                      warningMessage: missingExpectedAttributes.includes(inputTableData[rowIndex].field_name) ? 'This attribute doesn\'t exist in the data table' : ''
                     })],
                   ['object_builder',
                     () => {
-                      const selectedInputName = structInputDefinition[rowIndex].field_name
+                      const selectedInputName = inputTableData[rowIndex].field_name
                       const warningMessage = missingRequiredInputs.includes(selectedInputName) || missingExpectedAttributes.includes(selectedInputName) || inputsWithInvalidValues.includes(selectedInputName) ? 'One of this struct\'s inputs has an invalid configuration' : ''
 
                       return WithWarnings({
                         baseComponent: StructBuilderLink({
-                          onClick: () => setStructIndexPath([...structIndexPath, rowIndex])
+                          onClick: () => setStructIndexPath([...structIndexPath, inputTableData[rowIndex].configurationIndex])
                         }),
                         warningMessage
                       })
                     }],
                   ['none', () => WithWarnings({
                     baseComponent: h(TextCell,
-                      { style: Utils.inputTypeStyle(currentStructType.fields[rowIndex].field_type) },
-                      [isInputOptional(currentStructType.fields[rowIndex].field_type) ? 'Optional' : 'This input is required']
+                      { style: Utils.inputTypeStyle(inputTableData[rowIndex].field_type) },
+                      [isInputOptional(inputTableData[rowIndex].field_type) ? 'Optional' : 'This input is required']
                     ),
-                    warningMessage: missingRequiredInputs.includes(structInputDefinition[rowIndex].field_name) ? 'This attribute is required' : ''
+                    warningMessage: missingRequiredInputs.includes(inputTableData[rowIndex].field_name) ? 'This attribute is required' : ''
```

Comment:
Reminder to rebase and test with your other changes before merging (or vice versa if this merges first)